### PR TITLE
[#55] [Android] [UI] As a user, I can answer a net promoter score (NPS) question

### LINKED ISSUE #55: [Android] [UI] As a user, I can answer a net promoter score (NPS) question
# Why

User should be able to see and answer a net promoter score (NPS) question.

# Acceptance Criteria

- Show the answer range when question's `pick` is `one` and `display_type` is `nps`.
- Show `Not at all Likely` under first 4 values and `Extremely Likely` under last 4 values.

Image

[Figma](https://www.figma.com/file/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=48-79&t=1xiBQA7GNv0kEx7j-0)

### PR DESCRIPTION
**Pull request by AVI5HEK**
- Close #55 

## What happened 👀

- Show the answer range when question's `pick` is `one` and `display_type` is `nps`.
- Show `Not at all Likely` under first 4 values and `Extremely Likely` under last 4 values.

## Insight 📝

Added the composable.

## Proof Of Work 📹

image

### CODE REVIEW

**Review by suho — APPROVED** [review]

File: androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/NpsBar.kt [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        modifier = modifier
+            .wrapContentWidth()
+            .height(94.dp)
+    ) {
+        LazyRow(
+            modifier = Modifier
+                .border(BorderStroke(1.dp, White), MaterialTheme.shapes.medium)
+                .height(57.dp)
+        ) {
+            items(surveyAnswerUiModels.size) { index ->
```
Comment (kaungkhantsoe) [inline androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/NpsBar.kt]:
```suggestion
            items(
                count = surveyAnswerUiModels.size,
                key = { surveyAnswerUiModels[it].id },
            ) { index ->
```

File: androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/NpsBar.kt [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    var selectedIndex by remember { mutableStateOf(0) }
+    Box(
+        contentAlignment = Alignment.TopCenter,
+        modifier = modifier
+            .wrapContentWidth()
+            .height(94.dp)
+    ) {
+        LazyRow(
+            modifier = Modifier
+                .border(BorderStroke(1.dp, White), MaterialTheme.shapes.medium)
```
Comment (kaungkhantsoe) [inline androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/NpsBar.kt]:
```suggestion
                .border(
                    border = BorderStroke(1.dp, White),
                    shape = MaterialTheme.shapes.medium,
                )
```

File: androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/NpsBar.kt [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            )
+        }
+    }
+}
+
+@Preview
+@Composable
+fun NpsBarPreview() {
+    ApplicationTheme {
+        NpsBar(surveyAnswerUiModels = answerUiModels)
```
Comment (kaungkhantsoe) [inline androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/NpsBar.kt]: I think this is more appealing than reusing the `answerUiModels` and match the Figma design.
```suggestion
        NpsBar(surveyAnswerUiModels = List(10) {
            SurveyAnswerUiModel(
                id = (it + 1).toString(),
                text = "${it + 1}",
                displayOrder = it,
                placeholder = ""
            )
        })
```

File: androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/NpsBar.kt (hunk 1 above)
Comment (AVI5HEK) [reply androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/NpsBar.kt]: Updated in https://github.com/nimblehq/avishek-phong-kmm-ic/pull/126/commits/32e874347181ddd2ff1acf387ad8cbd887b1034e

File: androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/NpsBar.kt (hunk 2 above)
Comment (AVI5HEK) [reply androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/NpsBar.kt]: Updated in https://github.com/nimblehq/avishek-phong-kmm-ic/pull/126/commits/32e874347181ddd2ff1acf387ad8cbd887b1034e

File: androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/NpsBar.kt (hunk 3 above)
Comment (AVI5HEK) [reply androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/NpsBar.kt]: Updated in https://github.com/nimblehq/avishek-phong-kmm-ic/pull/126/commits/32e874347181ddd2ff1acf387ad8cbd887b1034e

**Review by kaungkhantsoe — APPROVED** [review]

**Pull request merged**